evalml/tests/component_tests/decomposer_tests/test_decomposer.py

```diff
@@ -462,6 +468,7 @@ def test_decomposer_determine_periodicity(
         period,
         trend_degree=trend_degree,
     )
+    y = ww.init_series(y.astype(int), logical_type=y_logical_type)
```

Comment:
I recognize that it feels very wrong that we can just truncate y by converting to int and still have the tests work.

Originally, I was scaling up by a factor of 1000 or so to try and maintain the information necessary to achieve periodicity, but that ended up with the [tests](https://github.com/alteryx/evalml/actions/runs/4495269110/jobs/7908723125?pr=4105) where the period is set to None and the trend_degree is 1 failing. I don't really know why that would cause those tests to fail but assumed it might have something to do with larger values interacting with the `acf_threshold`? Oddly, I found that changing the scaling factor to `10` caused more tests to fail but just truncating the data allowed all of them to pass. 

This is rubbing me the wrong way, but I don't really understand what's going on in `determine_periodicity` well enough to find a smarter way around this. I'm open to any thoughts anyone may have!

Reply:
@tamargrey which logical types fail if just using the original float information?

Reply:
I think I'm OK with this - an alternative would be to run the tests with the integer logical types with integer y by changing `generate_seasonal_data` but it would serve the same purpose. Wonder what @eccabay and @chukarsten think!

Reply:
I don't think I'm quite following why this is necessary. Is this just because we get errors when trying to convert the output of `generate_seasonal_data` into the correct logical type, or is some part of the actual function failing when we have non-integer data? With the former I'm unconcerned, but the latter would worry me.

Reply:
The first one - the output of `generate_seasonal_data` is floats, which we can't convert to `Int64` because pandas decided it will no longer quietly truncate your data for you as it does if the intended dtype is `int64`. 

I just needed a way to test `Int64` data with `determine_periodicity`, so rather than reinvent the wheel, I figured I could turn the output of `generate_seasonal_data` into something I could use. 

And to be clear: the incompatibility with `Int64` in `determine_periodicity` is due only to nullable type incompatibilities. It can handle `int64` data just fine.

Reply:
In that case, this feels fine to me. Thanks for the clarification!